python/ray/train/lightning/lightning_trainer.py

```diff
@@ -0,0 +1,422 @@
+import ray
+from inspect import isclass
+from typing import Any, Dict, Optional, Type
+
+from ray.air import session
+from ray.air.config import DatasetConfig, RunConfig, ScalingConfig
+from ray.air.checkpoint import Checkpoint
+from ray.data.preprocessor import Preprocessor
+from ray.train.trainer import GenDataset
+from ray.train.torch import TorchTrainer
+from ray.train.torch.config import TorchConfig
+from ray.util import PublicAPI
+from ray.train.lightning._lightning_utils import (
+    RayDDPStrategy,
+    RayEnvironment,
+    RayDataModule,
+)
+
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import ModelCheckpoint
+from pytorch_lightning.plugins.environments import ClusterEnvironment
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+@PublicAPI(stability="alpha")
+class LightningConfigBuilder:
+    """Configuration Class to pass into LightningTrainer.
+
+    Example:
+        .. code-block:: python
```

Comment:
Can we test this code snippet now that the api has been finalized?

Reply:
Just replace code-block with test-code

Reply:
Yes. My plan is to create a separate PR for release tests. I've tested on workspace that multi-node multi-machine training works. Also I wrote some ci tests with single-node multi-gpu setting and it also works. 
I'll post a PR tomorrow.

Reply:
What about for the docstring? Can we make sure to test those, now that the API is finalized?

Reply:
For the code in docstring, I'll test it as well, together with the release test PR.